src/controller/InvokeInteraction.h

```diff
@@ -48,7 +48,7 @@ InvokeCommandRequest(Messaging::ExchangeManager * aExchangeMgr, const SessionHan
                      const RequestObjectT & requestCommandData,
                      typename TypedCommandCallback<typename RequestObjectT::ResponseType>::OnSuccessCallbackType onSuccessCb,
                      typename TypedCommandCallback<typename RequestObjectT::ResponseType>::OnErrorCallbackType onErrorCb,
-                     const Optional<uint16_t> & timedInvokeTimeoutMs)
+                     const Optional<uint16_t> & timedInvokeTimeoutMs, const Optional<System::Clock::Timeout> & responseTimeout)
```

Comment:
Should that new arg, and others being added, default to NullOptional?

Reply:
Any reason this couldn't just be a vanilla `System::Clock::Timeout` defaulted to 0, where 0 would mean 'use default'?

0 isn't a legal value I believe to use with CommandSender anyways, so this might make this call site a bit leaner.

Reply:
I'd rather leave it as an optional so it's obvious. The magic default always feels a bit wrong. OK with you to leave this as optional with a NullOptional default?